File: app/schemas/category.py
# app/schemas/category.py

from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class CategoryBase(BaseModel):
    name: str
    icon: Optional[str] = None
    color: Optional[str] = None
    keywords: Optional[str] = None  # 逗号分隔的关键词


class CategoryCreate(CategoryBase):
    pass


class CategoryUpdate(BaseModel):
    name: Optional[str] = None
    icon: Optional[str] = None
    color: Optional[str] = None
    keywords: Optional[str] = None


class CategoryResponse(CategoryBase):
    id: int
    created_at: datetime

    class Config:
        from_attributes = True


class MatchRequest(BaseModel):
    text: str


class MatchResponse(BaseModel):
    matched: bool
    category: Optional[CategoryResponse] = None

File: web/src/types/category.ts
/** 对齐后端 app/schemas/category.py */

export interface CategoryCreate {
  name: string;
  icon?: string | null;
  color?: string | null;
  keywords?: string | null;
}

export interface CategoryUpdate {
  name?: string | null;
  icon?: string | null;
  color?: string | null;
  keywords?: string | null;
}

export interface CategoryResponse {
  id: number;
  name: string;
  icon: string | null;
  color: string | null;
  keywords: string | null;
  created_at: string;
}

export interface MatchRequest {
  text: string;
}

export interface MatchResponse {
  matched: boolean;
  category: CategoryResponse | null;
}
